core/src/utils/config.ts

```diff
@@ -2,6 +2,26 @@ import type { AnimationBuilder, Mode, SpinnerTypes, TabButtonLayout } from '../i
 
 import type { PlatformConfig } from './platform';
 
+export interface BaseComponentsConfig {
+  /**
+   * If provided, the specified components
+   * will be opted-in to base components. All
+   * other components will not be opted in.
+   * Developers should use either `includeComponent`
+   * or `excludeComponents` but not both.
+   */
+  includeComponents?: string[];
```

Comment:
~ Thinking of ergonomics of developers consuming this, thoughts on having the component tag names explicitly typed? This would provide auto completion and type checking against these options. 

Unsure if there is a generated type we could lean on here from Stencil's compiled output (maybe `IntrinsicElements`?). 

e.g.:

```ts
export type ComponentOption = "ion-button" | "ion-checkbox" | "ion-toggle";

export interface BaseComponentsConfig {
  includeComponents?: ComponentOption[];
  excludeComponents?: ComponentOption[];
}
```

Thoughts?

Reply:
I like that idea. I talked with Ryan, and no such interface exists at the moment. I will discuss this feature a bit more with them. Maybe we stick with `string[]` for now and then revamp it when the type is added?